app/controllers/redirects_controller.rb

```diff
@@ -0,0 +1,44 @@
+class RedirectsController < ApplicationController
+  layout 'dashboards'
+
+  skip_before_action :authenticate_user!
+  helper_method :school_redirect_path
+
+  def school_page_redirect
+    new_session and return if current_user.nil?
+
+    if current_user.group_admin? || (current_user.school_admin? && current_user.has_other_schools?)
+      choose_school
+    elsif current_user.admin?
+      redirect_to school_redirect_path(School.data_enabled.sample, params[:path]),
```

Comment:
wonder if showing choose_school for all schools would be better?

Reply:
That would be a lot of schools to show... but do see your point. We compromised on putting a notice up showing the admin user had been redirected to a random school.